### PR TITLE
LPD-18466 - Create new field to avoid concurrency issues during Salesforce integration tests

### PR DESCRIPTION
**Pull request by G-Sa**
Related to: [LPD-18466](https://liferay.atlassian.net/browse/LPD-18466)

### CODE REVIEW

File: modules/apps/object/object-storage-salesforce-test/src/testIntegration/java/com/liferay/object/storage/salesforce/internal/rest/manager/v1_0/test/SalesforceObjectEntryManagerImplTest.java [hunk 1]
```diff
@@ -168,6 +169,22 @@ public void setUp() throws Exception {
...
 
+		ObjectFieldUtil.addCustomObjectField(
+			new LongIntegerObjectFieldBuilder(
+			).externalReferenceCode(
+				"Object_Definition_id__c"
+			).userId(
+				adminUser.getUserId()
+			).labelMap(
+				LocalizedMapUtil.getLocalizedMap("Object Definition Id")
+			).listTypeDefinitionId(
```
Comment (guilhermedcamacho) [inline modules/apps/object/object-storage-salesforce-test/src/testIntegration/java/com/liferay/object/storage/salesforce/internal/rest/manager/v1_0/test/SalesforceObjectEntryManagerImplTest.java:180]: I think we don't need to fill **listTypeDefinitionId** for LongInteger field type.

File: modules/apps/object/object-storage-salesforce-test/src/testIntegration/java/com/liferay/object/storage/salesforce/internal/rest/manager/v1_0/test/SalesforceObjectEntryManagerImplTest.java [hunk 2]
```diff
@@ -445,7 +470,8 @@ protected Page<ObjectEntry> getObjectEntries(
 	}
 
 	private ObjectEntry _addObjectEntry(
-			String customStatus, Date date, String title)
+			String customStatus, Date date, String title,
+			long objectDefinitionId)
```
Comment (guilhermedcamacho) [inline modules/apps/object/object-storage-salesforce-test/src/testIntegration/java/com/liferay/object/storage/salesforce/internal/rest/manager/v1_0/test/SalesforceObjectEntryManagerImplTest.java:474]: Can you sort these parameters?

Comment (guilhermedcamacho) [reply modules/apps/object/object-storage-salesforce-test/src/testIntegration/java/com/liferay/object/storage/salesforce/internal/rest/manager/v1_0/test/SalesforceObjectEntryManagerImplTest.java:474]: What do you think about overloading this method, so you don't have to change all the invocations?

**Review by guilhermedcamacho — CHANGES_REQUESTED** [review]

File: modules/apps/object/object-storage-salesforce-test/src/testIntegration/java/com/liferay/object/storage/salesforce/internal/rest/manager/v1_0/test/SalesforceObjectEntryManagerImplTest.java (hunk 2 above)
Comment (G-Sa) [reply modules/apps/object/object-storage-salesforce-test/src/testIntegration/java/com/liferay/object/storage/salesforce/internal/rest/manager/v1_0/test/SalesforceObjectEntryManagerImplTest.java:474]: I fear that if we don't change all the invocations we might run into the same issue of concurrency

Comment (guilhermedcamacho) [reply modules/apps/object/object-storage-salesforce-test/src/testIntegration/java/com/liferay/object/storage/salesforce/internal/rest/manager/v1_0/test/SalesforceObjectEntryManagerImplTest.java:474]: My idea is to overload this method, so you would have these two private methods:

`private ObjectEntry _addObjectEntry(
			String customStatus, Date date, String title) {
			
			this._addObjectEntry(customStatus, date, title, _objectDefinition.getObjectDefinitionId())`
			
`private ObjectEntry _addObjectEntry(
			String customStatus, Date date, String title,
			long objectDefinitionId)`
			
The existing method calls the new one passing the default objectDefinitionId value. For our current tests we don't need to pass a different objectDefinitionId in these calls, so we have two options for developers, one using the default value and another if they want to pass a value other than that. Does it make sense?

**Comment by G-Sa** [comment]
ci:help

**Comment by G-Sa** [comment]
ci:test:sf

File: modules/apps/object/object-storage-salesforce-test/src/testIntegration/java/com/liferay/object/storage/salesforce/internal/rest/manager/v1_0/test/SalesforceObjectEntryManagerImplTest.java (hunk 2 above)
Comment (guilhermedcamacho) [reply modules/apps/object/object-storage-salesforce-test/src/testIntegration/java/com/liferay/object/storage/salesforce/internal/rest/manager/v1_0/test/SalesforceObjectEntryManagerImplTest.java:473]: It's much better now, but I think you forgot to sort the parameters.

**Review by guilhermedcamacho — CHANGES_REQUESTED** [review]

File: modules/apps/object/object-storage-salesforce-test/src/testIntegration/java/com/liferay/object/storage/salesforce/internal/rest/manager/v1_0/test/SalesforceObjectEntryManagerImplTest.java (hunk 2 above)
Comment (G-Sa) [reply modules/apps/object/object-storage-salesforce-test/src/testIntegration/java/com/liferay/object/storage/salesforce/internal/rest/manager/v1_0/test/SalesforceObjectEntryManagerImplTest.java:473]: I did, my bad!

**Review by guilhermedcamacho — APPROVED** [review]
LGTM! :tada:

**Comment by G-Sa** [comment]
ci:forward

**Comment by carolmariaabb** [comment]
ci:forward

**Pull request closed without merging**